pennylane_lightning/core/src/simulators/lightning_tensor/cutn/MPSCutn.hpp

```diff
@@ -0,0 +1,341 @@
+            if (i == 0) {
+                // Leftmost site (state mode, shared mode)
+                localSiteExtents = std::vector<std::size_t>(
+                    {BaseType::getQubitDims()[i], maxBondDim_});
+            } else if (i == BaseType::getNumQubits() - 1) {
+                // Rightmost site (shared mode, state mode)
+                localSiteExtents = std::vector<std::size_t>(
+                    {maxBondDim_, BaseType::getQubitDims()[i]});
+            } else {
+                // Interior sites (state mode, state mode, shared mode)
+                localSiteExtents = std::vector<std::size_t>(
+                    {maxBondDim_, BaseType::getQubitDims()[i], maxBondDim_});
+            }
+            localSitesExtents.push_back(std::move(localSiteExtents));
+        }
+        return localSitesExtents;
+    }
+
+    /**
+     * @brief Return siteExtents_int64 to the member initializer
+     *
+     * @return const std::vector<std::vector<int64_t>>
+     */
+    const std::vector<std::vector<int64_t>> setSitesExtents_int64_() {
+        std::vector<std::vector<int64_t>> localSitesExtents_int64;
+
+        for (std::size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            // Convert datatype of sitesExtents to int64 as required by
+            // cutensornet backend
+            std::vector<int64_t> siteExtents_int64(sitesExtents_[i].size());
+            std::transform(sitesExtents_[i].begin(), sitesExtents_[i].end(),
+                           siteExtents_int64.begin(), [](std::size_t x) {
+                               return static_cast<int64_t>(x);
+                           });
+
+            localSitesExtents_int64.push_back(std::move(siteExtents_int64));
+        }
+        return localSitesExtents_int64;
+    }
+
+    /**
+     * @brief The tensors init helper function for ctor.
+     */
+    void initTensors_() {
+        for (std::size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            // construct mps tensors reprensentation
+            tensors_.emplace_back(sitesModes_[i].size(), sitesModes_[i],
+                                  sitesExtents_[i], BaseType::getDevTag());
+        }
+    }
+
+    /**
+     * @brief Update quantumState (cutensornetState_t) with data provided by a
+     * user
+     *
+     * @param extentsIn Extents of each sites
+     * @param tensorsIn Pointer to tensors provided by a user
+     */
+    void updateQuantumStateMPS_(const int64_t *const *extentsIn,
+                                uint64_t **tensorsIn) {
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetStateInitializeMPS(
+            /*const cutensornetHandle_t*/ BaseType::getCutnHandle(),
+            /*cutensornetState_t*/ BaseType::getQuantumState(),
+            /*cutensornetBoundaryCondition_t*/
+            CUTENSORNET_BOUNDARY_CONDITION_OPEN,
+            /*const int64_t *const*/ extentsIn,
+            /*const int64_t *const*/ nullptr,
+            /*void **/ reinterpret_cast<void **>(tensorsIn)));
```

Comment:
Is the `/*void **/` comment meant to be `/*void ** */`, or is it fine as is?

Reply:
Nice spot!